### ISSUE TITLE
Add "Start the pipeline on source code change" to Pipeline manifest

### ISSUE THREAD

**Issue by JonBernardHealia** [aws/copilot-cli#4763]
With my project, I'd like to be able to manually trigger releases in my copilot pipeline. The CodePipeline interface allows me to toggle off `Change detection options` >> `Start the pipeline on source code change`, but that option does not appear to be available in the copilot pipeline manifest. So every time I deploy a change to the pipeline, I need to go back into the CodePipeline interface and toggle it back off.

Would be super useful if I could just set it in the manifest. I could change my ci pipeline to watch a different branch, but it'd be nice if Copilot didn't dictate my branch structure.

**Comment by dannyrandall** [comment]
Hey @JonBernardHealia! Thanks for the feature request.☺️ Just to clarify, you'd like the ability to, from the pipeline manifest, stop the pipeline from running on commits? Just want to make sure I understand correctly.😊

**Comment by JonBernardHealia** [comment]
Yes. I'm triggering the pipeline manually from Github Actions (for instance on tag creation) and would like to set this flag from the Copilot Pipeline manifest instead of having to do it through the UI.
